applications/StructuralMechanicsApplication/tests/adjoint_sensitivity_analysis_test_factory.py

```diff
@@ -0,0 +1,260 @@
+from __future__ import print_function, absolute_import, division #makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+import os
+
+#import kratos core and applications
+import KratosMultiphysics
+import KratosMultiphysics.StructuralMechanicsApplication as StructuralMechanicsApplication
+import KratosMultiphysics.KratosUnittest as KratosUnittest
+from KratosMultiphysics.StructuralMechanicsApplication import structural_mechanics_analysis
+import KratosMultiphysics.kratos_utilities as kratos_utilities
+from KratosMultiphysics import IsDistributedRun
+
+has_hdf5_application = kratos_utilities.CheckIfApplicationsAvailable("HDF5Application")
+
+class AdjointSensitivityAnalysisTestFactory(KratosUnittest.TestCase):
+    def setUp(self):
+        # Within this location context:
+        with KratosUnittest.WorkFolderScope(".", __file__):
+            # Reading the ProjectParameters
+            with open(self.primal_file_name,'r') as parameter_file:
+                self.primal_parameters = KratosMultiphysics.Parameters(parameter_file.read())
+            with open(self.adjoint_file_name,'r') as parameter_file:
+                self.adjoint_parameters = KratosMultiphysics.Parameters(parameter_file.read())
+            self.problem_name = self.primal_parameters["problem_data"]["problem_name"].GetString()
+            self.model_part_name = self.primal_parameters["solver_settings"]["model_part_name"].GetString()
+
+            # To avoid many prints
+            if (self.primal_parameters["problem_data"]["echo_level"].GetInt() == 0 or self.adjoint_parameters["problem_data"]["echo_level"].GetInt() == 0):
+                KratosMultiphysics.Logger.GetDefaultOutput().SetSeverity(KratosMultiphysics.Logger.Severity.WARNING)
+
+            self.verify_linear_solver(self.primal_parameters)
+            self.verify_linear_solver(self.adjoint_parameters)
+
+            # solve primal problem
+            model_primal = KratosMultiphysics.Model()
+            primal_analysis = structural_mechanics_analysis.StructuralMechanicsAnalysis(model_primal, self.primal_parameters)
+            primal_analysis.Run()
+            # create adjoint analysis
+            model_adjoint = KratosMultiphysics.Model()
+            self.adjoint_analysis = structural_mechanics_analysis.StructuralMechanicsAnalysis(model_adjoint, self.adjoint_parameters)
+            self.adjoint_analysis.Initialize()
+
+    def verify_linear_solver(self, parameter_file):
```

Comment:
Minor: is it possible to reuse this function from the SturcutMechFactory somehow? Maybe also slightly modifying the implementation there?

Reply:
good hint. I modified it.